### PR TITLE
refactor route handling [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
in prep for DataBiosphere/firecloud-app#199.

No new functionality in this PR - this is purely a refactor, moving code out of index.js into standalone include/requires.

DataBiosphere/firecloud-app#199 will require functional change to these same places and really wants this refactor; I am doing this as separate PRs to keep them small, manageable, and reviewable.

### CODE REVIEW

File: function/index.js [hunk 1]
```diff
@@ -11,91 +11,6 @@ const corsOptions = {
...
-            reqinfo.accepted = !!accepted;
-        }
-        return reqinfo;
-    }
-}
-
-function throwResponseError(statusCode, message) {
-    throw new ResponseError(message, statusCode);
-}
-
```
Comment (davidangb) [inline function/index.js]: this block moved to validation.js

File: function/index.js [hunk 2]
```diff
@@ -130,36 +45,8 @@ function tosapi(req, res, authClient, datastoreClient) {
...
-                        });
-                } catch (err) {
-                    return prefixedRejection(err, 'Error writing user response');
-                }
-            } else {
-                return Promise.reject({statusCode: 405});
-            }
-        });
+    // TODO: route to different handlers for userRequest vs. status
+    return toshandler.handleRequest(req, authorizer, datastore);
```
Comment (davidangb) [inline function/index.js]: this block moved to toshandler.js

File: function/toshandler.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const validation = require('./validation.js');
```
Comment (zarsky-broad) [inline function/toshandler.js]: Can you destructure this?

Comment (davidangb) [reply function/toshandler.js]: I certainly could ... 
```
const {validateRequestUrl, validateRequestMethod, validateContentType,
    requireAuthorizationHeader, validateInputs} = require('./validation.js');
```
felt more verbose to me. Is there a shorthand for that?

Comment (zarsky-broad) [reply function/toshandler.js]: No, but it saves you calling them each as `validation.foo` every time.

**Pull request merged**